File: TextIndexApi/src/main/java/com/hideakin/textsearch/index/model/AuthenticateErrorResponse.java
package com.hideakin.textsearch.index.model;

import com.hideakin.textsearch.index.data.AuthenticateError;

public class AuthenticateErrorResponse extends ErrorResponse {

	public AuthenticateErrorResponse() {
		super(AuthenticateError.INVALID_REQUEST, AuthenticateError.INVALID_REQUEST_DESC);
	}

	public AuthenticateErrorResponse(String error, String errorDescription) {
		super(error, errorDescription);
	}

	public AuthenticateErrorResponse(String error, String errorDescription, String errorUri) {
		super(error, errorDescription, errorUri);
	}

}

File: TextSearchCLI/model/AuthenticateErrorResponse.cs
using Newtonsoft.Json;

namespace com.hideakin.textsearch.model
{
    internal class AuthenticateErrorResponse
    {
        [JsonProperty("error")]
        public string Error { get; set; }

        [JsonProperty("error_description")]
        public string ErrorDescription { get; set; }

        [JsonProperty("error_uri")]
        public string ErrorUri { get; set; }
    }
}
